### ISSUE TITLE
Image captions or titles and acessibility

### ISSUE THREAD

**Issue by Yan-Thomas** [withastro/docs#560]
For now, AFAIK, we only have one image as part of Astro Docs, specifically, on the [Partial Hydration page](https://docs.astro.build/en/core-concepts/partial-hydration/), but in the future, we may want to add more visual explanations and two things crossed my mind:

- The current diagram image on Partial Hydration only has "diagram" as its alt text. This is not very descriptive and screen readers users would benefit from more detailed alt texts.

- Since the images are in English, different languages can't do much about it if not adding an extra explanation below the image or add an alt text. Would it be interesting if we added captions to images via a remark/rehype plugin, or inserted a title attribute based on its alt text?

**Comment by sarah11918** [comment]
I am absolutely +1 to alt-text. I can write some alt-text for the image that is there, and that can trigger translations. (I also think it deserve a proper credit. I know we link to the article above, but we don't explicitly say that this image is from that article.)

I'll make those changes for this one, and if you'd like to explore visible caption options, that would be great!

Note: I would like the option to NOT have (visible) captions on the images, so they should be optional. For docs, it's better practice to write your text in such a way that the image itself isn't necessary to get your point across, but instead is just a visual representation of what you've written. So, a text caption underneath the image won't always be helpful, might be distracting even. (And, in such a case, the alt-text *might* even be null, if it only interrupts someone on a screen reader because it's just "image to show what was already explained.")

**Comment by delucis** [comment]
Do we even need that image? The text doesn't really discuss what it shows very directly. It's complementary but doesn't use the same terminology talking about "apps" where the text talks about "islands".

**Comment by sarah11918** [comment]
I'll be brutal with it when I look at it. I've seen other partial hydration/Island architecture diagrams in other articles. I'll see whether one fits better, or consider making one myself, or just not having one entirely.

**Comment by sarah11918** [comment]
Alright, just an update here. I submitted a PR with better alt-text and a source for the image we have.

Since we have no other images, this isn't an *immediate* concern, and I would not add a new image without alt-text going forward. As for plugin/captions... how much do we want to keep this issue open for that right now? Do we want to close for now, while it's not a current *problem* and raise this again if/when it comes up?

**Comment by Yan-Thomas** [comment]
Thanks, @sarah11918!

IMO, I think we can close this. The caption/plugin idea is more an extra thing for the real problem which would be images without proper text explaining them and/or alt text. Just maybe documenting somewhere that all images **must have** explanations and/or alt text would be sufficient and we wouldn't need to complicate the docs authoring experience adding plugins, etc.

**Comment by sarah11918** [comment]
Great, then I'll close for now, and when we start adding diagrams, we'll revisit how to handle images in future!

**Comment by delucis** [comment]
For reference, the ever resourceful @hippotastic found this solution: https://deploy-preview-571--astro-docs-2.netlify.app/de/core-concepts/partial-hydration/#insel-architektur

He’s currently refactoring it into a reusable component so all languages could use it and just provide their own labels.

**Comment by hippotastic** [comment]
Refactoring is done, I'm waiting for the PR to get merged. See it in action here:
https://deploy-preview-571--astro-docs-2.netlify.app/de/core-concepts/partial-hydration/

**Comment by sarah11918** [comment]
@hippotastic That looks great! 

We should still keep some kind of line attributing credit though, since it's a a pretty close reproduction of the image?

**Comment by hippotastic** [comment]
The article already attributes Jason Miller's Islands Architecture article and even contains a link to it in the blockquote. The previously used image didn't feature any additional attribution whatsoever.

**Comment by sarah11918** [comment]
Yes, and I *just* fixed that!  ;)  We attributed the quote, but not the image. I updated last night.  :)

**Comment by hippotastic** [comment]
Ah, I see! I'll have a look what you did and add the attribution, too.

**Comment by hippotastic** [comment]
Done. I'll also update the English attribution line as soon as my PR got merged as it's no longer an "image" in our new version. And the alt text you added is also no longer applicable then because my version is fully accessible. 😅
